File: Week 4/PRG_레벨2_H-Index/hyungjoon_solved.py
'''
문제 : H_Index
난이도 : 레벨 2
소요 시간 : 50분
'''
def solution(citations):
    answer = 0
    n = len(citations)
    citations.sort(reverse=True)

    # h번 이상 인용된 논문이 현재 인덱스보다 많다면 조건에 부합한다. 최대값만 구하면 되니까 그 아래는 볼 필요가 없음.
    # 5 4 3 3 1 을 예시로 보면 i = 3, citations[i] = 3 으로 적합
    # 근데 5 4 3 2 1 을 보면 i = 2, citations[i] = 3 인데, 정답은 3임
    for i in range(n):
        if i >= citations[i]:
            return i
    
    return n

File: Week 17/PRG_레벨1_데이터 분석/hyungjoon_solved.java
import java.util.*;

/**
 * 문제 : 데이터 분석
 * 난이도 : 레벨 1
 * 소요 시간 : 24분
 */
class Solution {
    public static List<int[]> list = new ArrayList<>();
    public static Map<String, Integer> map = new HashMap<>();

    public int[][] solution(int[][] data, String ext, int val_ext, String sort_by) {
        // ext에 따른 데이터 맵핑
        map.put("code", 0);
        map.put("date", 1);
        map.put("maximum", 2);
        map.put("remain", 3);
        int extIdx = map.get(ext);
        int sortBy = map.get(sort_by);

        // 조건에 맞는 배열을 list로 전달
        for(int[] arr : data){
            int temp = arr[extIdx];
            if(temp < val_ext){
                list.add(arr);
            }
        }

        // list 정렬
        list.sort(Comparator.comparingInt(arr -> arr[sortBy]));

        // list를 2차원배열로 변환
        int[][] answer = new int[list.size()][4];
        for(int i = 0; i < list.size(); i++){
            answer[i] = list.get(i);
        }

        return answer;
    }
}